client/src/App.tsx

```diff
@@ -425,251 +415,263 @@ const App = () => {
-        args={args}
-        setArgs={setArgs}
-        sseUrl={sseUrl}
-        setSseUrl={setSseUrl}
-        env={env}
-        setEnv={setEnv}
-        bearerToken={bearerToken}
-        setBearerToken={setBearerToken}
-        onConnect={connectMcpServer}
-        stdErrNotifications={stdErrNotifications}
-        logLevel={logLevel}
-        sendLogLevelRequest={sendLogLevelRequest}
-        loggingSupported={!!serverCapabilities?.logging || false}
-      />
-      <div className="flex-1 flex flex-col overflow-hidden">
-        <div className="flex-1 overflow-auto">
-          {mcpClient ? (
-            <Tabs
-              defaultValue={
-                Object.keys(serverCapabilities ?? {}).includes(
-                  window.location.hash.slice(1),
-                )
-                  ? window.location.hash.slice(1)
-                  : serverCapabilities?.resources
-                    ? "resources"
-                    : serverCapabilities?.prompts
-                      ? "prompts"
-                      : serverCapabilities?.tools
-                        ? "tools"
-                        : "ping"
-              }
-              className="w-full p-4"
-              onValueChange={(value) => (window.location.hash = value)}
-            >
-              <TabsList className="mb-4 p-0">
-                <TabsTrigger
-                  value="resources"
-                  disabled={!serverCapabilities?.resources}
-                >
-                  <Files className="w-4 h-4 mr-2" />
-                  Resources
-                </TabsTrigger>
-                <TabsTrigger
-                  value="prompts"
-                  disabled={!serverCapabilities?.prompts}
-                >
-                  <MessageSquare className="w-4 h-4 mr-2" />
-                  Prompts
-                </TabsTrigger>
-                <TabsTrigger
-                  value="tools"
-                  disabled={!serverCapabilities?.tools}
-                >
-                  <Hammer className="w-4 h-4 mr-2" />
-                  Tools
-                </TabsTrigger>
-                <TabsTrigger value="ping">
-                  <Bell className="w-4 h-4 mr-2" />
-                  Ping
-                </TabsTrigger>
-                <TabsTrigger value="sampling" className="relative">
-                  <Hash className="w-4 h-4 mr-2" />
-                  Sampling
-                  {pendingSampleRequests.length > 0 && (
-                    <span className="absolute -top-1 -right-1 bg-red-500 text-white text-xs rounded-full h-4 w-4 flex items-center justify-center">
-                      {pendingSampleRequests.length}
-                    </span>
+  if (window.location.pathname === "/oauth/callback") {
+    const OAuthCallback = React.lazy(
+      () => import("./components/OAuthCallback"),
+    );
+    return (
+      <Suspense fallback={<div>Loading...</div>}>
+        <OAuthCallback />
+      </Suspense>
+    );
+  } else {
```

Comment:
this `else` is redundant here because of the `return` above

Reply:
Not really. It could be that the diff view was confusing. This is the structure we're looking at:

<img width="546" alt="Screenshot 2025-03-28 at 3 16 48 PM" src="https://github.com/user-attachments/assets/0e7b6800-9248-47bf-9ec4-11cc931721e3" />

The return above the `} else {` only executes if that condition was met, otherwise we return the component after the `} else {`.

Reply:
Wouldn't this also work:

```js
if(condition) {
 return (...)
}
return (...)
```

instead of 
```js
if(condition) {
 return (...)
} else {
 return (...)
}
```

Reply:
It would, but the else makes it clear that what's returned from the component is either or. It's not less performant or incorrect. If I were to revisit, I'd make those blocks functions and have a single return statement at the end for clarity. Two returns is an anti-pattern.

Reply:
Alright 👍, I was just thinking that removing that `else`  would help in reducing the indentation or the diff size of the entire block.

Reply:
True that. Easier to approve. I'll do that.

Reply:
@pulkitsharma07 Ok, that did make the diff much shorter, so it should be easier to be confident about the changes.